### ISSUE TITLE
irc support in im breaks base isupport feature of IRCClient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6263]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6263|
|Type|defect|
|Created|2013-01-18 20:26:17Z|
|Branch|https://github.com/twisted/twisted/tree/ircsupport-isupport-6263|

`twisted.words.im.ircsupport.IRCProto` overrides `connectionMade` and does not call the base implementation.

When _ISUPPORT_ support was added, the base implementation began initializing the `supported` attribute in `connectionMade`.  Parsing of later messages from the server depends on this attribute having been initialized.

The result is that when `IRCProto` receives an _ISUPPORT_ message, the parsing code it inherits from `IRCClient` fails with an `AttributeError`.  This failure is logged and the _ISUPPORT_ information is lost.

<details><summary>Searchable metadata</summary>

```
trac-id__6263 6263
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_ircsupport_isupport_6263 branches/ircsupport-isupport-6263
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1358540777000000 1358540777000000
changetime__1358552756000000 1358552756000000
version__None None
owner__exarkun exarkun
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36851]) Branching to 'ircsupport-isupport-6263'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [36852]) Add a test for IRCProto handling of ISUPPORT and fix it by making IRCProto.connectionMade call the base implementation

refs [#6263](https://github.com/twisted/twisted/issues/6263)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ircsupport-isupport-6263)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to @exarkun|
|-|-|

This looks good, please merge.

Note: The `'PASS :secret'` -> `'PASS :secret'` change looks strange, but [RFC 1459](http://www.irc.org/tech_docs/rfc1459.html) suggests the original code was simply bogus.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [36853]) Merge ircsupport-isupport-6263

Author: exarkun
Reviewer: tom.prince
Fixes: [#6263](https://github.com/twisted/twisted/issues/6263)

Be sure to call the base implementation (`IRCClient.connectionMade`) from the im-specialized
IRC client (`twisted.words.im.ircsupport.IRCProto`) to benefit from the initialization done
there, both so that future ISUPPORT-related parsing works and to avoid duplicating the
login logic already available from the base class.

This also slightly changes how im sends the _PASS_ command, but does so by
deleting duplicate code produces a result more likely to be correct.
